lib/sqitch-config.pod

```diff
@@ -484,6 +484,25 @@ by C<core.top_dir>.
 The file name extension on deploy, revert, and verify SQL scripts. Defaults to
 C<sql>.
 
+=item C<core.strict>
```

Comment:
Appreciate the docs here, but I think the option and config need to be documented for every command they apply to.  Should include, at least, these files, I think:

```
lib/sqitch-deploy.pod
lib/sqitch-revert-usage.pod
lib/sqitch-rebase.pod
lib/sqitch-revert.pod
lib/sqitch-deploy-usage.pod
lib/sqitch-checkout-usage.pod
lib/sqitch-checkout.pod
lib/sqitch-rebase-usage.pod
lib/App/Sqitch/Command/revert.pm
lib/App/Sqitch/Command/deploy.pm
```

The main pod files for each command also include a "Configuration Variables" that should have the strict mode documented -- anywhere the string `no_prompt` appears.

Reply:
I'm happy to extend the documentation of course, but in some of the cases that you listed I'm not sure what I would say.
* `sqitch-deploy.pod`, `sqitch-deploy-usage.pod`, and `deploy.pm`: I don't think this change affects the deploy command at all?
* `sqitch-deploy-usage.pod`, `sqitch-checkout-usage.pod`, `sqitch-revert-usage.pod`, and `sqitch-rebase-usage.pod`: These files don't seem to currently describe configuration options at all. Are you proposing to add new sections?
* `deploy.pm` and `revert.pm`: It looks to me like these are only documenting the attributes, not the configuration, is that wrong? I don't think this PR changes any of the attributes for these two modules.

>The main pod files for each command also include a "Configuration Variables" that should have the strict mode documented -- anywhere the string no_prompt appears.

I think this is covered by your suggestions above, but let me know if you disagree.

I will push a change to `sqitch-rebase.pod`, `sqitch-revert.pod`, `sqitch-checkout.pod` in the meantime.

Reply:
* Right, not deploy, sorry. 
* I see options listed in the `-usage.pod` files; new ones  like `--strict` that apply to each command should be added (this is related to my other request not to make it a global option I guess).
* Yeah, if you move the option to `revert.pm` it would then need to be documented.

Reply:
Oh, I don't think we are adding new options here, only new configuration variables.

Unless there is something which I don't understand that automatically creates options from configuration settings?

Reply:
Oh, right. In that case it should probably follow the cascading configuration pattern of

1. `--strict`
2. `$cmd.strict`
3. `target.$name.strict`
4. `engine.$name.strict`
5. `core.strict`

Most of that resolution is done in App::Sqitch::Target, using the `_fetch` method:

https://github.com/sqitchers/sqitch/blob/aace1e837b5e07fa0d8cfb924505b221f58079c4/lib/App/Sqitch/Target.pm#L76-L84

I think it'd be fine to do so in a subsequent PR.